### ISSUE TITLE
"go install https://github.com/CtrlSpice/otel-desktop-viewer" fails for latest version

### ISSUE THREAD

**Issue by jmorrell** [CtrlSpice/otel-desktop-viewer#117]
https://github.com/CtrlSpice/otel-desktop-viewer/pull/107 introduced a `replace` directive to support factoring out the desktop exporter as its own installable go module so people can directly use it in their collector configs.

That was always meant to be the case, however this has a drawback that `go install https://github.com/CtrlSpice/otel-desktop-viewer@version` no longer works.

```shell
go install github.com/CtrlSpice/otel-desktop-viewer@99b375929f66321ded1fee68421c398c2dfea41f
go: downloading github.com/CtrlSpice/otel-desktop-viewer v0.1.2-0.20230508163713-99b375929f66
go: github.com/CtrlSpice/otel-desktop-viewer@99b375929f66321ded1fee68421c398c2dfea41f (in github.com/CtrlSpice/otel-desktop-viewer@v0.1.2-0.20230508163713-99b375929f66):
	The go.mod file for the module providing named packages contains one or
	more replace directives. It must not contain directives that would cause
	it to be interpreted differently than if it were the main module.
```

This seems to be a fairly common issue https://github.com/golang/go/issues/44840, and I've not been able to find a way that enables splitting the modules **and** also the same `go install` one-line install we've been using.

We can still give a copy-pasteable set of commands that would work:

```shell
git clone git@github.com:CtrlSpice/otel-desktop-viewer.git
cd otel-desktop-viewer
go install
```

However this would always install `main`. To specify a specific version is a little more verbose

```shell
git clone git@github.com:CtrlSpice/otel-desktop-viewer.git
cd otel-desktop-viewer
git checkout v0.1.1
go install
```

This also prints out a bunch of text which would likely be scary to new developers:

```
❯ git checkout v0.1.1
Note: switching to 'v0.1.1'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 4917612 Add per platform executables (#103)
```

One way around this would be to split the project into two repos with the `desktopexporter` as a separate project that gets pulled into a `otel-desktop-viewer` cli distribution. I think that may make sense in the long-term, especially if the project is adopted upstream into the OpenTelemetry project, however my preference would be to avoid this complexity for now.

Another way would be to publish the binaries we are are already producing for the releases to various package managers. The goreleaser tool we are already using has some support for this. We could likely crib from [`otel-cli`'s configuration](https://github.com/equinix-labs/otel-cli/blob/main/.goreleaser.yml) which does much of this already. This would allow users to `brew install otel-desktop-viewer`.

**Comment by codeboten** [comment]
An alternative would be to remove the replace statement altogether and just set the version to main. This may makes things a bit more annoying for development, but i think in the end the user experience is better.

Using goreleaser to publish binaries and use that for end users would likely be an improvement as well. In the short term maybe just remove the replace statement and follow up with another issue to publish binaries?

**Comment by codeboten** [comment]
This is now fixed in main, i updated the dependency to use what's in main

```
 v0.0.0-20230515221800-9c594f0581e6
```

Once the next release is made, go install should work again.

**Comment by notcool11** [comment]
Not sure if related but I get this error when attempting the install:
```
go % go install github.com/CtrlSpice/otel-desktop-viewer@latest
go: github.com/CtrlSpice/otel-desktop-viewer@latest (in github.com/CtrlSpice/otel-desktop-viewer@v0.1.2):
	The go.mod file for the module providing named packages contains one or
	more replace directives. It must not contain directives that would cause
	it to be interpreted differently than if it were the main module.
```

**Comment by codeboten** [comment]
This is now fixed w/ the v0.1.4 tag
